src/compiler/emitter.ts

```diff
@@ -1693,8 +1693,20 @@ module ts {
             }
 
             function isExistingName(location: Node, name: string) {
-                return !resolver.isUnknownIdentifier(location, name) ||
-                    (currentScopeNames && hasProperty(currentScopeNames, name));
+                if (!resolver.isUnknownIdentifier(location, name)) {
+                    return true;
+                }
+                if (currentScopeNames && hasProperty(currentScopeNames, name)) {

```

Comment:
resolver can tell you about names that you wrote, however it has no idea about names that were generated by the emitter - this is why we need to consult both resolver and chain of name scopes in emitter